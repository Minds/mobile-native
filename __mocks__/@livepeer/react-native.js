// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
