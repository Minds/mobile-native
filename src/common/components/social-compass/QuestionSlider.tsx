import React from 'react';
import { View } from 'react-native';
import Slider from '@react-native-community/slider';
import { Slider as ValueSlider } from 'react-native-elements';
import MText from '~/common/components/MText';
import ThemedStyles from '~/styles/ThemedStyles';

import i18n from '../../services/i18n.service';
import { IQuestion } from './useQuestions';
import { IS_IOS } from '~/config/Config';

const THUMB_TOUCH_SIZE = {
  width: 80,
  height: 80,
};

type PropsType = {
  question: IQuestion;
  onAnswer: (number) => void;
};

const QuestionSlider = ({ question, onAnswer }: PropsType) => {
  const slider = IS_IOS ? (
    <Slider
      style={styles.barView}
      thumbTintColor={ThemedStyles.getColor('Link')}
      value={
        typeof question.currentValue === 'number'
          ? question.currentValue
          : question.defaultValue
      }
      onSlidingComplete={onAnswer}
      minimumValue={question.minimumRangeValue || 0}
      step={question.stepSize}
      maximumValue={question.maximumRangeValue || 100}
      maximumTrackTintColor={ThemedStyles.getColor('PrimaryBorder')}
      minimumTrackTintColor={ThemedStyles.getColor('PrimaryBorder')}
    />
  ) : (
    <ValueSlider
      value={
        typeof question.currentValue === 'number'
          ? question.currentValue
          : question.defaultValue
      }
      onSlidingComplete={onAnswer}
      thumbTintColor={ThemedStyles.getColor('Link')}
      minimumValue={question.minimumRangeValue || 0}
      allowTouchTrack={false}
      step={question.stepSize}
      trackStyle={trackStyle}
      maximumValue={question.maximumRangeValue || 100}
      thumbTouchSize={THUMB_TOUCH_SIZE}
      maximumTrackTintColor={ThemedStyles.getColor('PrimaryBorder')}
      minimumTrackTintColor={ThemedStyles.getColor('PrimaryBorder')}
    />
  );

  return (
    <View style={styles.container}>
      <MText style={styles.title}>
        {i18n.tf(
          `socialCompass.customizeQuestions.${question.questionId}.title`,
          question.questionText,
        )}
      </MText>
      {slider}
      <View style={styles.textContainer} pointerEvents="none">
        <MText style={styles.text}>
          {i18n.tf(
            `socialCompass.customizeQuestions.${question.questionId}.leftText`,
            question.minimumStepLabel,
          )}
        </MText>
        <MText style={styles.text}>
          {i18n.tf(
            `socialCompass.customizeQuestions.${question.questionId}.rightText`,
            question.maximumStepLabel,
          )}
        </MText>
      </View>
    </View>
  );
};

const styles = ThemedStyles.create({
  container: ['marginBottom7x'],
  title: ['fontL', 'fontMedium'],
  textContainer: ['rowJustifySpaceBetween'],
  text: [{ fontSize: 13 }, 'fontMedium', 'colorSecondaryText'],
});

const trackStyle = { height: 1 };

export default QuestionSlider;
