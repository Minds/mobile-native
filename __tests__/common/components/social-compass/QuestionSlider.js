import React from 'react';
import 'react-native';
import { render } from '@testing-library/react-native';
import QuestionSlider from '../../../../src/common/components/social-compass/QuestionSlider';
import sp from '~/services/serviceProvider';

jest.mock('~/services/serviceProvider');

// mock services
sp.mockService('styles');
sp.mockService('i18n');

const mockData = {
  questionText: 'questionText',
  questionId: 'questionId',
  stepSize: 1,
  defaultValue: 24,
  currentValue: 25,
  maximumRangeValue: 0,
  minimumRangeValue: 100,
  minimumStepLabel: 'minimumStepLabel',
  maximumStepLabel: 'maximumStepLabel',
};

describe('Slider component', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<QuestionSlider question={mockData} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
