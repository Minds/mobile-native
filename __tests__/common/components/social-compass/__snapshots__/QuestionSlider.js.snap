// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 28,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
        },
        Object {
          "fontFamily": "Roboto",
        },
        Array [
          Object {
            "fontSize": 16,
          },
          Object {
            "fontFamily": "Roboto-Medium",
            "fontWeight": "500",
          },
        ],
      ]
    }
  >
    questionText
  </Text>
  <View
    allowTouchTrack={false}
    animationType="timing"
    onLayout={[Function]}
    step={1}
    style={
      Object {
        "height": 40,
        "justifyContent": "center",
      }
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
    thumbTouchSize={
      Object {
        "height": 80,
        "width": 80,
      }
    }
    value={25}
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#414C57",
          "borderRadius": 2,
          "height": 1,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#414C57",
          "borderRadius": 2,
          "height": 0,
          "marginTop": -0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#1B85D6",
          "borderRadius": 10,
          "height": 20,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 20,
        }
      }
      testID="sliderThumb"
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
          },
          Object {
            "fontFamily": "Roboto",
          },
          Array [
            Object {
              "fontSize": 13,
            },
            Object {
              "fontFamily": "Roboto-Medium",
              "fontWeight": "500",
            },
            Object {
              "color": "#B8C1CA",
            },
          ],
        ]
      }
    >
      minimumStepLabel
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
          },
          Object {
            "fontFamily": "Roboto",
          },
          Array [
            Object {
              "fontSize": 13,
            },
            Object {
              "fontFamily": "Roboto-Medium",
              "fontWeight": "500",
            },
            Object {
              "color": "#B8C1CA",
            },
          ],
        ]
      }
    >
      maximumStepLabel
    </Text>
  </View>
</View>
`;
